### PR TITLE
fix: make the examples consistent with the changes in #2553

### PR DESCRIPTION
**Pull request by ManasviGoyal**
I noticed that the header-only C++ examples didn't reflect the changes made in the C++ LayoutBuilder API in #2553. Fixed the code to add the required changes.

### CODE REVIEW

**Comment by agoose77** [comment]
Thanks Manasvi! We really should add a test that these compile, let me modify our CI.

File: header-only/examples/cython/demo_impl.cpp [hunk 1]
```diff
@@ -72,8 +72,8 @@ ArrayBuffers create_demo_array() {
                     NumpyBuilder<int32_t>>>
     > builder(fields_map);
 
-    auto &one_builder = builder.field<Field::one>();
-    auto &two_builder = builder.field<Field::two>();
+    auto &one_builder = builder.content<Field::one>();
```
Comment (ianna) [inline header-only/examples/cython/demo_impl.cpp:75]: @ManasviGoyal - thanks for spotting this. Indeed we have changed the API of the builders.

**Review by ianna — APPROVED** [review]
Good catch! Thanks.

**Review by jpivarski — APPROVED** [review]
Thanks for catching this! When everyone is in agreement, let's merge it.

File: header-only/examples/cython/CMakeLists.txt [hunk 2]
```diff
@@ -18,11 +18,9 @@ if(NOT awkward-headers_POPULATED)
                    EXCLUDE_FROM_ALL)
 endif()
 
+find_package(Python REQUIRED COMPONENTS Interpreter Development.Module NumPy)
```
Comment (agoose77) [inline header-only/examples/cython/CMakeLists.txt:21]: This is not a required change (besides linking NumPy), but it's more modern.

**Review by agoose77 — APPROVED** [review]

**Pull request merged**